app/src/main/java/com/arctouch/codechallenge/ui/base/BaseViewModel.kt

```diff
@@ -0,0 +1,36 @@
+package com.arctouch.codechallenge.ui.base
+
+import android.widget.Toast
+import androidx.annotation.StringRes
+import androidx.databinding.ObservableBoolean
+import androidx.lifecycle.ViewModel
+import com.arctouch.codechallenge.App
+import io.reactivex.disposables.CompositeDisposable
+import timber.log.Timber
+
+interface IViewModelCycle {
```

Comment:
Hello, about this... This is something that i choose to do in projects that i work alone, because i think this is more clean than have your concrete classes with impl or implementation at the end. I do care about the code glance, and i think putting impl or implementation make your code ugly. Just my opinion.